### PR TITLE
Remove `bundle install --path` and print an error if `bundle install --path` is used

### PR DESCRIPTION
**Pull request by smlance**

### CODE REVIEW

File: lib/bundler/cli/install.rb [hunk 1]
```diff
@@ -44,9 +44,15 @@ def run
...
-                         "as well as --system. Please choose."
+      if options[:path]
+        Bundler.ui.error "You have specified an installation path with the "\
+         "path flag. Please use `bundle config path #{options[:path]}` instead."
+        exit 1
+      end
+
+      if (Bundler.settings[:path] || options[:deployment]) && options[:system]
+        Bundler.ui.error "You have configured a path to install your gems to, \n" \
+                         "and specified the --system path. Please choose."
```
Comment (segiddins) [inline lib/bundler/cli/install.rb]: Please only use one?

File: lib/bundler/source/rubygems.rb [hunk 2]
```diff
@@ -130,6 +130,9 @@ def install(spec, opts = {})
             bin_path     = Bundler.system_bindir
           end
 
+          $stderr.puts "install_path: #{install_path}"
```
Comment (segiddins) [inline lib/bundler/source/rubygems.rb]: delete these

**Comment by smlance** [comment]
Specs fail at the moment (local failures here https://gist.github.com/smlance/7be52e21745eae585271); it's possible this is related to https://github.com/bundler/bundler/commit/3202988684406fbd2f03a83c281fb9811d3d2c9a.

File: lib/bundler/cli/install.rb (hunk 1 above)
Comment (smlance) [inline lib/bundler/cli/install.rb]: Forgot about that -- thanks!

File: lib/bundler/source/rubygems.rb (hunk 2 above)
Comment (smlance) [inline lib/bundler/source/rubygems.rb]: Absolutely. This is a WIP PR.

File: lib/bundler/cli/install.rb (hunk 1 above)
Comment (smlance) [inline lib/bundler/cli/install.rb]: Actually, why? They're distinct errors.

Comment (segiddins) [inline lib/bundler/cli/install.rb]: I meant changing `Please choose` to `Please only use one`

**Comment by smlance** [comment]
The specs should pass now, but this needs to be rebased onto 2-0-dev.

**Comment by indirect** [comment]
@smlance can you use git to squash this down to a single commit?

**Comment by smlance** [comment]
Coming right up!

**Comment by segiddins** [comment]
:+1: . @indirect r?

**Comment by indirect** [comment]
@homu r+

**Pull request merged**